Content.Server/WebPlacer/WebPlacerSystem.cs

```diff
@@ -0,0 +1,102 @@
+using Content.Server.Popups;
+using Content.Shared.Maps;
+using Content.Shared.WebPlacer;
+using Content.Shared.Whitelist;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+
+namespace Content.Server.WebPlacer;
+
+/// <summary>
+/// Spawns entities (probably webs) around the component owner when using the component's action.
+/// </summary>
+/// <seealso cref="WebPlacerComponent"/>
+public sealed class WebPlacerSystem : SharedWebPlacerSystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+    [Dependency] private readonly ITileDefinitionManager _tile = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedMapSystem _map = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<WebPlacerComponent, SpiderWebActionEvent>(OnSpawnWeb);
+    }
+
+    private void OnSpawnWeb(Entity<WebPlacerComponent> webPlacer, ref SpiderWebActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var xform = Transform(webPlacer.Owner);
+        var grid = xform.GridUid;
+
+        // Instantly fail in space.
+        if (!TryComp<MapGridComponent>(grid, out var gridComp))
+        {
+            _popup.PopupEntity(Loc.GetString(webPlacer.Comp.MessageOffGrid), args.Performer, args.Performer);
+            return;
+        }
+
+        // Get the spawn locations.
+        List<EntityCoordinates> spawnPos = new();
+        foreach (var vect in webPlacer.Comp.OffsetVectors)
+            spawnPos.Add(xform.Coordinates.Offset(vect));
+
+        bool success = false;
+        foreach (var pos in spawnPos)
+        {
```

Comment:
Ed is a little overzealous with the "no linq" thing (Sloth said it's fine to use select here), but frankly you dont even need a second loop here to map the values.

This runs in fewer cycles and uses less memory. Win-win!

```suggestion
        bool success = false;
        foreach (var vect in webPlacer.Comp.OffsetVectors)
        {
            var pos = xform.Coordinates.Offset(vect);
```

Reply:
In the back of my mind I knew I didn't need two loops, but I just didn't think too hard about it.